pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/ProtoParserTools.java

```diff
@@ -211,7 +214,16 @@ public static String readString(final ReadableSequentialData input) {
             throw new BufferUnderflowException();
         }
 
-        return new String(bytes,StandardCharsets.UTF_8);
+        try {
+            // Shouldn't use `new String()` because we want to error out on malformed UTF-8 bytes.
+            return StandardCharsets.UTF_8.newDecoder()
+                    .onMalformedInput(CodingErrorAction.REPORT)
+                    .onUnmappableCharacter(CodingErrorAction.REPORT)
+                    .decode(ByteBuffer.wrap(bytes))
```

Comment:
This, unfortunately, creates an extra ByteBuffer object. I can't think of how to avoid it, though

Reply:
True. I haven't found a public API that would allow me to decode bytes and throw exceptions w/o me having to create a ByteBuffer. Let's see if Oleg discovers any performance degradation because of this. If so, then the only other option would be to cut'n'paste the code from the OpenJDK String implementation, which I'd prefer not to do unless we have to.

Reply:
The `new String` alternative used previously _copies_ the array, so I would argue that this new approach is significantly less pressure on memory.  ByteBuffer.wrap allocates a HeapByteBuffer that uses the existing array as its backing store, so it's a pretty small allocation by comparison.
If you want to avoid the call to `wrap`, I believe the following will work:
Instead of creating the byte[] on line 211, you could create a `ByteBuffer` there and use `long readBytes(@NonNull final ByteBuffer dst)` on the `ReadableSequentialData`.  Then you don't allocate the array separately, just the ByteBuffer that's needed here.  It doesn't avoid the ByteBuffer or reduce total allocations, but does make the code a little tidier.

Reply:
That's an awesome suggestion, @jsync-swirlds . Thanks!
I just implemented it and pushed it into this branch.